drivers/SmartThings/zwave-switch/src/zooz-zen-30-dimmer-relay/init.lua

```diff
@@ -0,0 +1,151 @@
+  for _, fingerprint in ipairs(ZOOZ_ZEN_30_DIMMER_RELAY_FINGERPRINTS) do
+    if device:id_match(fingerprint.mfr, fingerprint.prod, fingerprint.model) then
+      return true
+    end
+  end
+  return false
+end
+
+local BUTTON_VALUES = {
+  "up_hold", "down_hold", "held",
+  "up","up_2x","up_3x","up_4x","up_5x",
+  "down","down_2x","down_3x","down_4x","down_5x",
+  "pushed", "pushed_2x","pushed_3x","pushed_4x","pushed_5x"
+}
+
+local map_key_attribute_to_capability = {
+  [CentralScene.key_attributes.KEY_PRESSED_1_TIME] = {
+    [0x01] = capabilities.button.button.up(),
+    [0x02] = capabilities.button.button.down(),
+    [0x03] = capabilities.button.button.pushed()
+  },
+  [CentralScene.key_attributes.KEY_PRESSED_2_TIMES] = {
+    [0x01] = capabilities.button.button.up_2x(),
+    [0x02] = capabilities.button.button.down_2x(),
+    [0x03] = capabilities.button.button.pushed_2x()
+  },
+  [CentralScene.key_attributes.KEY_PRESSED_3_TIMES] = {
+    [0x01] = capabilities.button.button.up_3x(),
+    [0x02] = capabilities.button.button.down_3x(),
+    [0x03] = capabilities.button.button.pushed_3x()
+  },
+  [CentralScene.key_attributes.KEY_PRESSED_4_TIMES] = {
+    [0x01] = capabilities.button.button.up_4x(),
+    [0x02] = capabilities.button.button.down_4x(),
+    [0x03] = capabilities.button.button.pushed_4x()
+  },
+  [CentralScene.key_attributes.KEY_PRESSED_5_TIMES] = {
+    [0x01] = capabilities.button.button.up_5x(),
+    [0x02] = capabilities.button.button.down_5x(),
+    [0x03] = capabilities.button.button.pushed_5x()
+  },
+  [CentralScene.key_attributes.KEY_HELD_DOWN] = {
+    [0x01] = capabilities.button.button.up_hold(),
+    [0x02] = capabilities.button.button.down_hold(),
+    [0x03] = capabilities.button.button.held()
+  }
+}
+
+local function endpoint_to_component(device, endpoint)
+  if endpoint == 1 then
+    return "switch1"
+  elseif endpoint == 0 then
+    return "main"
+  end
+end
+
+local function component_to_endpoint(device, component)
+  if component == "switch1" then
+    return {1}
+  elseif component == "main" then
+    return {0}
+  end
+end
+
+local function central_scene_notification_handler(driver, device, cmd)
+  if(cmd.args.key_attributes == 0x01) then
+    log.error("Button Value 'released' is not supported by SmartThings")
+    return
+  end
+
+  if device:get_field(LAST_SEQ_NUMBER_KEY) ~= cmd.args.sequence_number then
+    device:set_field(LAST_SEQ_NUMBER_KEY, cmd.args.sequence_number)
+    local event_map = map_key_attribute_to_capability[cmd.args.key_attributes]
+    local event = event_map and event_map[cmd.args.scene_number]
+    if event ~= nil then
+      device:emit_event_for_endpoint("main", event)
+    end
+  end
+end
+
+local function added_handler(self, device)
+  device:emit_event(capabilities.button.supportedButtonValues(BUTTON_VALUES))
+  device:emit_event(capabilities.button.numberOfButtons({value = 3}))
+end
+
+local function device_init(self, device)
+  device:set_field(LAST_SEQ_NUMBER_KEY, -1)
```

Comment:
I don't think this needs to be initialized, and if you don't initialize it, you can avoid having to re-set the endpoint/component mapping functions

Reply:
> I don't think this needs to be initialized, and if you don't initialize it, you can avoid having to re-set the endpoint/component mapping functions 

I think we need the endpoint/component functions because of having 2 components (main and switch1)

Reply:
the default function in the base driver already does this:

```
--- Map component to end_points(channels)
---
--- @param device st.zwave.Device
--- @param component_id string ID
--- @return table dst_channels destination channels e.g. {2} for Z-Wave channel 2 or {} for unencapsulated
local function component_to_endpoint(device, component_id)
  local ep_num = component_id:match("switch(%d)")
  return { ep_num and tonumber(ep_num) }
end

--- Map end_point(channel) to Z-Wave endpoint 9 channel)
---
--- @param device st.zwave.Device
--- @param ep number the endpoint(Z-Wave channel) ID to find the component for
--- @return string the component ID the endpoint matches to
local function endpoint_to_component(device, ep)
  local switch_comp = string.format("switch%d", ep)
  if device.profile.components[switch_comp] ~= nil then
    return switch_comp
  else
    return "main"
  end
end
```

however if you overwrite the init handler you have to redefine it, which is why I'm suggesting you avoid doing that

Reply:
ok, I deleted the re-set and change unit tests a bit